sourcegraph/learn
# Add dark mode

### What should this PR do?
Resolves [DEVED-218](https://sourcegraph.atlassian.net/browse/DEVED-218) by adding dark mode to the site.

Users should be able to toggle between modes using the button in the main nav (screenshots below); user browser settings should also be respected.
 
### Why are we making this change?
- Dark mode is fun & will engage users with the site.

### What are the acceptance criteria? 
- User browser settings are respected.
- Previous toggle selection (`localStorage`) is respected.
- Toggling between modes works.
- No flashes between dark/light mode (see this [PR](https://github.com/sourcegraph/sourcegraph/pull/14586) for some context on the issue).
- Styles look reasonable.
- Lighthouse contrasts are within bounds.

### How should this PR be tested?
- Check out branch and check that your settings are respected.
- Ensure that any previous selection is respected (`localStorage`).
- Ensure that toggle works.
- Ensure no flashes occur when loading, regardless of mode.
- Check that styles are reasonable.

## Pull request process

**Reviewers**:

1. Test functionality using the criteria above.
2. Offer tips for efficiency, feedback on best practices, and possible alternative approaches and things that may not have been considered.
3. For shorter, "quick" PRs, use your best judgement on #​2.
4. Use a collaborative approach and provide resources and/or context where appropriate.
5. Provide screenshots/grabs where appropriate to show findings during review.

**Reviewees**:

1. Prefer incremental and appropriately-scoped changes.
2. Leave a comment on things you want explicit feedback on.
3. Respond clearly to comments and questions.
